contracts/Comptroller.sol

```diff
@@ -1079,6 +1079,35 @@ contract Comptroller is ComptrollerV7Storage, ComptrollerInterface, ComptrollerE
         require(unitroller._acceptImplementation() == 0, "change not authorized");
     }
 
+    /// @notice Delete this function after proposal 65 is executed
+    function fixBadAccruals(address[] calldata affectedUsers, uint[] calldata amounts) external {
+        require(msg.sender == admin, "Only admin can call this function"); // Only the timelock can call this function
+        require(!proposal65FixExecuted, "Already executed this one-off function"); // Require that this function is only called once
+        require(affectedUsers.length == amounts.length, "Invalid input");
+
+        // Loop variables
+        address user;
+        uint currentAccrual;
+        uint amountToSubtract;
+
+        // Iterate through all affected users
+        for (uint i = 0; i < affectedUsers.length; ++i) {
+            user = affectedUsers[i];
+            currentAccrual = compAccrued[user];
+
+            amountToSubtract = amounts[i];
+            // Prevent subtraction underflow in the case where the user has claimed and received an incorrect amount of COMP.
```

Comment:
This is fine; I'd be curious if we actually want to keep a "subtractive adjustment" for each account in storage forever s.t. we can subtract it out for accounts that were overpaid. In that way, those accounts will effectively work off the over-payment as opposed to going to zero. This is probably overkill, but it would lead to the most correct accounting.

Just to be clear, I mean, if you accrued 5, but the bug lead you to accrue 500, then you'd have a -495 modifier and would currently have "0 - 495" as your comp accrual. So even if you earn 3 comp, then your comp accrual would be "3 - 495 = -492".

This isn't a big difference since those accounts would just move to fresh accounts, but I'm wondering if it makes a little more sense from an accounting perspective.

Reply:
Good point from an accounting perspective. I could add a mapping and an event or two.

I think it's best not to include this mapping in the calculations of claimable COMP b/c as you said, users would just use a new account. The end result would be higher gas costs to claim.